src/template.yml

```diff
@@ -30,48 +30,69 @@ Parameters:
   CrossAccountAccessRoleName:
     Type: String
     Default: OrganizationAccountAccessRole
+    AllowedPattern: "\\A[a-zA-Z0-9_+=,.@\\-]+\\Z"
 
   MainNotificationEndpoint:
     Type: String
     Default: ""
-    Description: "Example -> jane@example.com"
+    Description: >-
+      Example -> jane@example.com. (Only required when installing ADF for the
+      first time.)
 
   DeploymentAccountName:
     Type: String
     Default: ""
-    Description: "The name of the new or existing deployment account"
+    Description: >-
+      The name of the new or existing deployment account. (Only required when
+      installing ADF for the first time.)
 
   DeploymentAccountEmailAddress:
     Type: String
     Default: ""
-    Description: "The email address for the new or existing deployment account"
+    Description: >-
+      The email address for the new or existing deployment account. (Only required when
+      installing ADF for the first time.)
 
   DeploymentAccountAlias:
     Type: String
     Default: ""
-    Description: "(Optional) Example -> companyname-deployment (must be globally unique)"
+    AllowedPattern: "\\A([a-z0-9]([a-z0-9]|-(?!-))*[a-z0-9])?\\Z"
+    MinLength: 0
+    MaxLength: 63
+    Description: >-
+      Example -> companyname-deployment (Must be globally unique. Only required when
+      installing ADF for the first time.)
 
   DeploymentAccountId:
     Type: String
     Default: ""
     Description: >-
-      Example -> 123456789012  (Only required if you have an existing AWS Account
-      that you wish to use as the deployment account.)
+      Example -> 123456789012  (Only supported when installing ADF for the first time
+      and you have an existing AWS Account that you wish to use as the deployment
+      account. Leave blank otherwise.)
 
   DeploymentAccountMainRegion:
     Type: String
-    Default: ""
+    AllowedPattern: "\\A[a-z][a-z0-9\\-]*\\Z"
```

Comment:
This would also match `uswest22322fasddfds`

Consider: `(us(-gov)?|ap|ca|cn|eu|sa)-(central|(north|south)?(east|west)?)-\d`

https://www.regextester.com/109163

Reply:
p.s. I don't think you need `\\A` and `\\Z`, Cloud-formation doesn't do partial matching.

The pattern must match the entire parameter value provided.

Reply:
Thanks for that regex, I will update it.
I will remove cn from the region prefix options, as we don't support that yet. Otherwise it looks like we support it.